pkg/provider/aws/execution_unit.go

```diff
@@ -34,6 +34,30 @@ func (a *AWS) expandExecutionUnit(dag *core.ResourceGraph, unit *core.ExecutionU
 	return nil
 }
 
+func (a *AWS) getLambdaConfiguration(result *core.ConstructGraph, dag *core.ResourceGraph, refs []core.AnnotationKey) (resources.LambdaFunctionConfigureParams, error) {
+	if len(refs) != 1 {
+		return resources.LambdaFunctionConfigureParams{}, fmt.Errorf("lambda must only have one construct reference")
+	}
+	lambdaConfig := resources.LambdaFunctionConfigureParams{}
+	construct := result.GetConstruct(refs[0].ToId())
```

Comment:
Isn't this use-case what `MapResourceDirectlyToConstruct` was built for?

Reply:
thats to get the resource which the construct is tied to, but here its the opposite. We would have to loop over all the keys in the map (since the keys are the constructs) and check the ids of their value